setwd("/home/dell/Documents/koushiki/project/")

#to remove redundant genes, transcripts, and exons
data <- read.csv("gencode.v41.basic.annotation.txt", header = F, sep = "\t")
colnames(data) <- c('seqname', 'source', 'feature', 'start' , 'end', 'score', 'strand', 'frame', 'attribute')
head(data)
library(dplyr)
x <- data %>% filter(feature=="gene")
head(x)
count(x)
y <- data %>% filter(feature=="gene") %>% distinct(start, end, .keep_all = T)
count(y)
a <- data %>% filter(feature=="exon")
count(a)
b <- data %>% filter(feature=="exon") %>% distinct(start, end, .keep_all = T)
head(b)
count(b)
c <- data %>% filter(feature=="transcript")
count(c)
d <- data %>% filter(feature=="transcript") %>% distinct(start, end, .keep_all=T)
head(d)
count(d)
write.table(y, file="gene.txt", sep="\t", quote = F, row.names =F, col.names = F )
write.table(d, file="transcript.txt", sep="\t", quote = F, row.names =F, col.names = F )
write.table(b, file="exon.txt", sep="\t", quote = F, row.names =F, col.names = F )
